contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,198 @@
+                 * Put otoken net after expiry: same as above:
+                 *     (long cash value * long amount) - (short cash value * short amount)
+                 */
+                netOtoken = netOtokenAfterExpiry;
+            } else {
+                /*
+                 * Call otoken net after expiry: (net otoken value / underlying price)
+                 */
+                (uint256 underlyingPrice, ) = _getUnerlyingPrice(address(short));
+                // todo: do we need to check isFinalized?
+                int256 underlyingPirceInt = FixedPointInt256.uintToInt(underlyingPrice);
+                netOtoken = netOtokenAfterExpiry.div(underlyingPirceInt);
+            }
+        } else {
+            // If option is not expired yet.
+            int256 shortStrike = FixedPointInt256.uintToInt(OtokenInterface(short).strikePrice());
+
+            // long otoken strike price, set to 0 if there's not long assets.
+            int256 longStrike = 0;
+            if (_vault.longOtokens.length > 0)
+                longStrike = FixedPointInt256.uintToInt(OtokenInterface(_vault.longOtokens[0]).strikePrice());
+
+            if (isPut) {
+                /**
+                 * Net otoken value for put =
+                 *     (long strike * min (short amount, long amount)) - (short amount * short strike)
+                 */
+                netOtoken = longStrike.mul(FixedPointInt256.min(shortAmount, longAmount)).sub(
+                    shortAmount.mul(shortStrike)
+                );
+            } else {
+                /**
+                 * Net otoken value for call =
+                 *     Min(0, long amount - short amount)
+                 *         - Min(long amount, short amount) * Max(0, long strike - short strike) / long strike)
+                 */
+                if (longStrike == 0) {
+                    // no long otoken
+                    netOtoken = -shortAmount;
+                } else {
+                    netOtoken = FixedPointInt256.min(0, longAmount.sub(shortAmount)).sub(
+                        FixedPointInt256
+                            .min(longAmount, shortAmount)
+                            .mul(FixedPointInt256.max(0, longStrike.sub(shortStrike)))
+                            .div(longStrike)
+                    );
+                }
+            }
+        }
+
+        int256 net = collateralAmount.add(netOtoken);
+        isExcess = net >= 0;
+        netValue = net.intToUint();
+    }
+
+    /**
+     * @dev internal function to get underlying price of an otoken.
+     * @param _otoken otoken address
+     */
+    function _getUnerlyingPrice(address _otoken) internal view returns (uint256 price, bool isFinalized) {
+        OtokenInterface otoken = OtokenInterface(_otoken);
+        bytes32 batchId = _getBatchId(otoken);
+        OracleInterface oracle = OracleInterface(AddressBookInterface(addressBook).getOracle());
+        return oracle.getPrice(batchId, otoken.expiryTimestamp());
```

Comment:
it is `getBatchPrice()` in the Oracle contract not `getPrice()`

Reply:
fixed